traffic_ops/traffic_ops_golang/role/roles.go

```diff
@@ -273,3 +292,529 @@ RETURNING id, last_updated`
 func deleteQuery() string {
 	return `DELETE FROM role WHERE id = :id`
 }
+
+// Update will modify the role identified by the role name.
+func Update(w http.ResponseWriter, r *http.Request) {
+	var roleID int
+	var roleName string
+	var roleDesc string
+	var privLevel int
+	var roleCapabilities *[]string
+	var roleV4 tc.RoleV4
+	var role TORole
+	var ok bool
+	var err error
+
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	version := inf.Version
+	if version == nil {
+		middleware.NotImplementedHandler().ServeHTTP(w, r)
+		return
+	}
+	tx := inf.Tx.Tx
+	if version.Major >= 4 {
```

Comment:
Based on `routes.go`, this handler is **only** for APIv4 Roles. I think your changes in this file are building off of my branch, but in my branch I removed the old CRUDer stuff and used each method function as the entrypoint for all versions of that request method.

If you're going to leave the CRUDer in and still use it serve old API versions, then there's no reason to handle different API versions separately in these functions, because it actually only handles one API version.